src/overlays/actors/ovl_En_Zod/z_en_zod.c

```diff
@@ -28,62 +57,592 @@ ActorInit En_Zod_InitVars = {
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+
+    this->cymbalRots[0] = this->cymbalRots[1] = this->cymbalRots[2] = 0;
+    this->actor.gravity = this->actor.terminalVelocity = -4.0f;
+    this->cymbalRotVels[0] = this->cymbalRotVels[1] = this->cymbalRotVels[2] = 300;
+    this->stateFlags = 0;
+    this->nextAnimIndex = -1;
+    this->curAnimIndex = -1;
+    this->actor.textId = 0;
+    this->unk_298 = 0;
+
+    for (i = 0; i < ARRAY_COUNT(this->drumScales); i++) {
+        this->drumScales[i] = 0.0f;
+        this->drumScaleVels[i] = 0.01;
+    }
+
+    EnZod_ChangeAnim(this, ENZOD_ANIM_PLAYING_LENTO, ANIMMODE_ONCE);
+    this->actionFunc = EnZod_PlayDrumsSequence;
+
+    switch (ENZOD_GET_TYPE(thisx)) {
+        case 1:
+            if (gSaveContext.save.weekEventReg[78] & 1) {
+                this->actionFunc = func_80BAFDB4;
+                EnZod_ChangeAnim(this, ENZOD_ANIM_PLAYING_VIVACE, ANIMMODE_ONCE);
+                this->actor.flags |= ACTOR_FLAG_10;
+                ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+                break;
+            }
+
+            this->actionFunc = func_80BAFB84;
+            if (!(gSaveContext.save.weekEventReg[55] & 0x80)) {
+                Actor_Kill(&this->actor);
+                break;
+            }
+            break;
+
+        case 2:
+            this->actionFunc = func_80BAFF14;
+            this->fogNear = -1;
+            this->stateFlags |= TIJO_STATE_2;
+            break;
+
+        default:
+            if (gSaveContext.save.weekEventReg[55] & 0x80) {
+                Actor_Kill(&this->actor);
+            }
+            this->actor.flags |= ACTOR_FLAG_10;
+            break;
+    }
+}
+
+void EnZod_Destroy(Actor* thisx, PlayState* play) {
+    EnZod* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void EnZod_HandleRoomConversation(EnZod* this, PlayState* play) {
+    u16 textId;
+
+    if (gSaveContext.save.playerForm != PLAYER_FORM_ZORA) {
+        textId = 0x1227;
+        if (gSaveContext.save.weekEventReg[32] & 8) {
+            textId = 0x1229;
+        } else {
+            gSaveContext.save.weekEventReg[32] |= 8;
+        }
+    } else if (this->stateFlags & TIJO_STATE_1) {
+        textId = 0x1225;
+    } else {
+        textId = 0x1219;
+        if (gSaveContext.save.weekEventReg[32] & 0x10) {
+            textId = 0x1226;
+        } else {
+            gSaveContext.save.weekEventReg[32] |= 0x10;
+        }
+        this->stateFlags |= TIJO_STATE_1;
+    }
+
+    EnZod_ChangeAnim(this, ENZOD_ANIM_PLAYING_VIVACE, ANIMMODE_ONCE);
+    Message_StartTextbox(play, textId, &this->actor);
+}
+
+s32 EnZod_PlayerIsFacingTijo(EnZod* this, PlayState* play) {
+    if ((this->actor.playerHeightRel < 30.0f) && (this->actor.xzDistToPlayer < 200.0f) &&
+        Player_IsFacingActor(&this->actor, 0x3000, play) && Actor_IsFacingPlayer(&this->actor, 0x3000)) {
+        return true;
+    }
+    return false;
```

Comment:
```suggestion
        return true;
    } else {
        return false;
    }
```
looks better since the branches are very similar.